applications/nrf5340_audio/src/bluetooth/le_audio_cis_gateway.c

```diff
@@ -4,3 +4,399 @@
+		if (device_found(type, p_ad->data, len - 1, addr) == 0) {
+			return;
+		}
+
+		(void)net_buf_simple_pull(p_ad, len - 1);
+	}
+}
+
+static void on_device_found(const bt_addr_le_t *addr, int8_t rssi, uint8_t type,
+			    struct net_buf_simple *p_ad)
+{
+	/* Note: May lead to connection creation */
+	ad_parse(p_ad, addr);
+}
+
+static void start_scan(void)
+{
+	int ret;
+
+	/* This demo doesn't require active scan */
+	ret = bt_le_scan_start(BT_LE_SCAN_ACTIVE, on_device_found);
+	if (ret != 0) {
+		LOG_INF("Scanning failed to start: %d", ret);
+		return;
+	}
+
+	LOG_INF("Scanning successfully started");
+}
+
+static void connected(struct bt_conn *conn, uint8_t err)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	(void)bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	if (err != 0) {
+		LOG_INF("Failed to connect to %s: %d", addr, err);
+
+		bt_conn_unref(default_conn);
+		default_conn = NULL;
+
+		start_scan();
+		return;
+	}
+
+	if (conn != default_conn) {
+		return;
+	}
+
+	LOG_INF("Connected: %s", addr);
+	bt_conn_set_security(conn, BT_SECURITY_L2);
+}
+
+static void disconnected(struct bt_conn *conn, uint8_t reason)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	if (conn != default_conn) {
+		return;
+	}
+
+	(void)bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	LOG_INF("Disconnected: %s (reason 0x%02x)", addr, reason);
+
+	bt_conn_unref(default_conn);
+	default_conn = NULL;
+
+	start_scan();
+}
+
+static void security_changed_cb(struct bt_conn *conn, bt_security_t level, enum bt_security_err err)
+{
+	int ret;
+
+	if (err) {
+		LOG_ERR("Security failed: level %u err %d", level, err);
+		ret = bt_conn_disconnect(conn, err);
+		if (ret) {
+			LOG_ERR("Failed to disconnect %d", ret);
+		}
+	} else {
+		LOG_INF("Security changed: level %u", level);
+		static struct bt_audio_discover_params dis_params;
+
+		dis_params.func = discover_sink_cb;
+		dis_params.type = BT_AUDIO_SINK;
+
+		err = bt_audio_discover(default_conn, &dis_params);
+		if (err != 0) {
+			LOG_INF("Failed to discover sink: %d", err);
+		}
+	}
+}
+
+BT_CONN_CB_DEFINE(conn_callbacks) = {
+	.connected = connected,
+	.disconnected = disconnected,
+	.security_changed = security_changed_cb,
+};
+
+int le_audio_config_get(void)
+{
+	return 0;
+}
+
+int le_audio_volume_up(void)
+{
+	return 0;
+}
+
+int le_audio_volume_down(void)
+{
+	return 0;
+}
+
+int le_audio_volume_mute(void)
+{
+	return 0;
+}
+
+int le_audio_play(void)
+{
+	return 0;
+}
+
+int le_audio_pause(void)
+{
+	return 0;
+}
+
+int le_audio_send(uint8_t const *const data, size_t size)
+{
+	int ret;
+	struct net_buf *buf;
+
+	buf = net_buf_alloc(iso_tx_pools[0], K_NO_WAIT);
+	net_buf_reserve(buf, BT_ISO_CHAN_SEND_RESERVE);
+
+	net_buf_add_mem(buf, data, 120);
+
+	ret = bt_audio_stream_send(&audio_stream, buf);
+	if (ret < 0) {
+		LOG_WRN("Failed to send audio data: %d", ret);
+		net_buf_unref(buf);
+	}
+
+	return 0;
+}
+
+void le_audio_enable(le_audio_receive_cb recv_cb)
+{
+	audio_stream.ops = &stream_ops;
```

Comment:
Create a bool so that audio_stream.ops = &stream_ops; is only done the first time le_audio_enable() is called?

Reply:
Might need ARG_UNUSED(recv_cb)